src/main/java/org/concordion/internal/command/executeCommand/ExecuteCommand.java

```diff
@@ -17,6 +22,92 @@
     public ExecuteCommand() {
     }
 
+    @Override
+    public void modifyCommandCallTree(CommandCall commandCall, List<CommandCall> examples, List<CommandCall> beforeExamples) {
+        if (isTableElement(commandCall)) {
+            performTableModification(commandCall);
+        }
+
+        if (isListElement(commandCall)) {
+            performListModification(commandCall);
+        }
+    }
+
+    private void performListModification(CommandCall commandCall) {
+    }
+
+
+    private Map<Integer, CommandCall> populateCommandCallByColumnMap(Table table, CommandCall tableCommandCall) {
+        Map<Integer, CommandCall> commandCallByColumn = new HashMap<Integer, CommandCall>();
+
+        Row headerRow = table.getLastHeaderRow();
+        for (CommandCall childCall : tableCommandCall.getChildren().asCollection()) {
+            int columnIndex = headerRow.getIndexOfCell(childCall.getElement());
+            if (columnIndex == -1) {
+                throw new RuntimeException("Commands must be placed on <th> elements when using 'execute' or 'verifyRows' commands on a <table>.");
+            }
+            commandCallByColumn.put(columnIndex, childCall);
+        }
+
+        return commandCallByColumn;
+    }
+
+    private void performTableModification(CommandCall commandCall) {
+
+
+        /*
+
+        We have to:
+        * Copy the execute commandCall to each TR except for the header rows
+        * Copy the TH commandCall to each TD
+        * remove the table execute commandCall from the table
+        * remove the TH commandCalls from the TH rows.
+         */
+
+        Table table = new Table(commandCall.getElement());
+        Map<Integer, CommandCall> headerCommands = populateCommandCallByColumnMap(table, commandCall);
+
+        // copy the execute to each detail row.
+        Row headerRow = table.getLastHeaderRow();
+
+        for (Row row : table.getDetailRows()) {
+            CommandCall rowCommand = duplicateCommandForDifferentElement(commandCall, row.getElement());
+            rowCommand.setParent(commandCall);
+
+            for (int cellCount = 0; cellCount < row.getCells().length; cellCount++) {
+                CommandCall headerCall = headerCommands.get(cellCount);
+
+                if (headerCall != null) {
+                    Element cellElement = row.getCells()[cellCount];
+                    CommandCall cellCommand = duplicateCommandForDifferentElement(headerCall, cellElement);
+                    cellCommand.setParent(rowCommand);
+                }
+            }
+        }
+
+        for (CommandCall headerCommand : headerCommands.values()) {
+            headerCommand.setParent(null);
+        }
+    }
+
+    private CommandCall duplicateCommandForDifferentElement(CommandCall commandCall, Element element) {
+
+        String expression = commandCall.getExpression();
+        if (expression.equals("")) {
```

Comment:
Any way we can clean this up? It seems to only be needed for example commands, but hard to check what the command is here since it's wrapped in decorators.